src/ispc/kernels/lanczos3.ispc

```diff
@@ -90,13 +90,40 @@ struct SampleWeights {
     uniform const WeightCollection* horizontal_weights;
 };
 
+uint8<1> sample_1_channel(const uniform uint8* varying pixel_ptr) {
+    const uniform uint8<1>* pixel_ptr1 = (const uniform uint8<1>*)(pixel_ptr);
+    varying uint8<1> dst = {0};
+    dst = *pixel_ptr1;
+    return dst;
+}
+
+void clean_and_write_1_channel(varying float<1> color, uniform uint8* varying pixel_ptr) {
+    pixel_ptr[0] = clamp(color[0], 0.0f, 255.0f);
+}
+
+uint8<2> sample_2_channels(const uniform uint8* varying pixel_ptr) {
+    // Memory reinterpretation to read all channels at once rather than one-by-one.
+    // While testing, this proved more performant than reading one-by-one.
+    const uniform uint8<2>* pixel_ptr2 = (const uniform uint8<2>*)(pixel_ptr);
+    varying uint8<2> dst = {0, 0};
+    dst = *pixel_ptr2;
+    return dst;
+}
+
+void clean_and_write_2_channels(varying float<2> color, uniform uint8* varying pixel_ptr) {
+    // The final color is a sum of numbers that are multiplied by the weights of their respective pixels.
+    // Because of their numbers, floating point precision leads to the final color being potentially outside of the 0-255 range by a slight margin.
+    // This would cause an underflow/overflow, which we avoid with the clamps.
+    pixel_ptr[0] = clamp(color[0], 0.0f, 255.0f);
+    pixel_ptr[1] = clamp(color[1], 0.0f, 255.0f);
+}
+
 uint8<3> sample_3_channels(const uniform uint8* varying pixel_ptr) {
     // Memory reinterpretation to read all channels at once rather than one-by-one.
     // While testing, this proved more performant than reading one-by-one.
     const uniform uint8<3>* pixel_ptr3 = (const uniform uint8<3>*)(pixel_ptr);
     varying uint8<3> dst = {0, 0, 0};
     dst = *pixel_ptr3;
-    return dst;
```

Comment:
How is this function supposed to compile without a return?

Reply:
Ah! Bad rebase! Nice catch :)

Reply:
Could it - and the recompile of _all the precompiled files_ - have solved the segfault you were seeing?